pkg/kritis/cryptolib/verify_detached_test.go

```diff
@@ -0,0 +1,183 @@
+const ec384_sig = "MGUCMFMH3lMmZrM4qPIegBZxb64fBkfS4jCeZ5XCVw0BYqC27PdTjoTyLaLlBWJvLpS6hAIxAJWsDJLdrhNyVhp9zMLHtFB4J_Q_QrZNh2tflxtVNFwWU3_JDKBr4g7vQFNIfQ880A"
+const ec521_sig = "MIGIAkIAofN81k5oSaXMYtoClAYQyVNv2aN1jJtCoIJKeQ0x4bGZAZdpGX8TMdUbiOjfjOedkOE55i94qb4UXzyuvGT0OegCQgC6-8kEXa72KL9upGhzQRzoZOku0EsbyOkwQOHDtj-HZxUG-lGsBhQsc2ABqXoiK07ZmdvMd_t358oVKl_isjEx5w"
+
+const bad_key = "malformed key"
+
+func TestVerifyDetached(t *testing.T) {
+
+	tcs := []struct {
+		name          string
+		signature     []byte
+		pubkey        []byte
+		signingAlg    SignatureAlgorithm
+		plaintext     []byte
+		expectedError bool
+	}{
+		{
+			name:          "good RsaSignPkcs12048Sha256 signature",
+			signature:     []byte(rsa2048_256_sig),
+			pubkey:        []byte(rsa_2048_pub_key),
+			signingAlg:    RsaSignPkcs12048Sha256,
+			plaintext:     []byte(good_plaintext),
+			expectedError: false,
+		},
+		{
+			name:          "good RsaSignPkcs14096Sha512 signature",
+			signature:     []byte(rsa4096_512_sig),
+			pubkey:        []byte(rsa_4096_pub_key),
+			signingAlg:    RsaSignPkcs14096Sha512,
+			plaintext:     []byte(good_plaintext),
+			expectedError: false,
+		},
+		{
+			name:          "good RsaPss4096Sha512 signature",
+			signature:     []byte(rsa4096_512_pss_sig),
+			pubkey:        []byte(rsa_4096_pub_key),
+			signingAlg:    RsaPss4096Sha512,
+			plaintext:     []byte(good_plaintext),
+			expectedError: false,
+		},
+		{
+			name:          "good RsaPss2048Sha256 signature",
+			signature:     []byte(rsa2048_256_pss_sig),
+			pubkey:        []byte(rsa_2048_pub_key),
+			signingAlg:    RsaPss2048Sha256,
+			plaintext:     []byte(good_plaintext),
+			expectedError: false,
+		},
+		{
+			name:          "good EcdsaP256Sha256 signature",
+			signature:     []byte(ec256_sig),
+			pubkey:        []byte(ec256_pub_key),
+			signingAlg:    EcdsaP256Sha256,
+			plaintext:     []byte(good_plaintext),
+			expectedError: false,
+		},
+		{
+			name:          "good EcdsaP384Sha384 signature",
+			signature:     []byte(ec384_sig),
+			pubkey:        []byte(ec384_pub_key),
+			signingAlg:    EcdsaP384Sha384,
+			plaintext:     []byte(good_plaintext),
+			expectedError: false,
+		},
+		{
+			name:          "good EcdsaP521Sha512 signature",
+			signature:     []byte(ec521_sig),
+			pubkey:        []byte(ec521_pub_key),
+			signingAlg:    EcdsaP521Sha512,
+			plaintext:     []byte(good_plaintext),
+			expectedError: false,
+		},
+		{
+			name:          "bad pub key",
+			signature:     []byte(rsa2048_256_sig),
+			pubkey:        []byte(bad_key),
+			signingAlg:    RsaSignPkcs12048Sha256,
+			plaintext:     []byte(good_plaintext),
+			expectedError: true,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			decoded_sig, decodeErr := base64.RawURLEncoding.DecodeString(string(tc.signature))
+			if decodeErr != nil {
+				t.Errorf("error base64 decoding signature: %e", decodeErr)
```

Comment:
Fatalf, because the rest of the test is trying to use decoded_sig

Reply:
done.